archeryutils/rounds.py

```diff
@@ -160,11 +166,7 @@ def max_distance(self, unit: bool = False) -> Union[float, Tuple[float, str]]:
         """
         max_dist = 0.0
         for pass_i in self.passes:
-            dist = (
-                pass_i.distance / YARD_TO_METRE
-                if pass_i.native_dist_unit == "yard"
-                else pass_i.distance
-            )
+            dist = Length.from_metres(pass_i.distance, pass_i.native_dist_unit)
             if dist > max_dist:
```

Comment:
While checking the change here noticed a bug, max_distance is comparing across units. A round with mixed distance units will give the wrong result here (eg 75m > 80y). Not common, but will come up for anyone making custom rounds, or potentially in IFAA where feet and yards are mixed.

Reply:
Can include a test and fix in next commits, assuming this was undesired behaviour

Reply:
Nice spot, yes this should give the longest distance irrespective of unit of measurement.
Fix appreciated.